### ISSUE TITLE
Game crashes when helmet is placed on Lord of the Rings armor stand.

### ISSUE THREAD

**Issue by alray2569** [alray2569/SpartanMod#1]
When placing a helmet on a Lord of the Rings mod armor stand, the game crashes with the following error report.

```
---- Minecraft Crash Report ----
// Everything's going to plan. No, really, that was supposed to happen.
Time: 9/7/15 2:38 PM
Description: Rendering Block Entity
java.lang.NullPointerException: Rendering Block Entity
 at com.andrewlray.mcmods.andrew_spartanmod.items.FeatheredArmor.getArmorModel(FeatheredArmor.java:52)
 at net.minecraftforge.client.ForgeHooksClient.getArmorModel(ForgeHooksClient.java:313)
 at lotr.client.render.tileentity.LOTRRenderArmorStand.func_147500_a(LOTRRenderArmorStand.java:73)
 at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147549_a(SourceFile:100)
 at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147544_a(SourceFile:92)
 at net.minecraft.client.renderer.RenderGlobal.func_147589_a(RenderGlobal.java:483)
 at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1224)
 at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1011)
 at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001)
 at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
 at net.minecraft.client.main.Main.main(SourceFile:148)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:483)
 at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
 at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
-- Head --
Stacktrace:
 at com.andrewlray.mcmods.andrew_spartanmod.items.FeatheredArmor.getArmorModel(FeatheredArmor.java:52)
 at net.minecraftforge.client.ForgeHooksClient.getArmorModel(ForgeHooksClient.java:313)
 at lotr.client.render.tileentity.LOTRRenderArmorStand.func_147500_a(LOTRRenderArmorStand.java:73)
-- Block Entity Details --
Details:
 Name: LOTRArmorStand // lotr.common.tileentity.LOTRTileEntityArmorStand
 Block type: ID #426 (tile.lotr:armorStand // lotr.common.block.LOTRBlockArmorStand)
 Block data value: 1 / 0x1 / 0b0001
 Block location: World: (166679,112,-6740), Chunk: (at 7,7,12 in 10417,-422; contains blocks 166672,0,-6752 to 166687,255,-6737), Region: (325,-14; contains chunks 10400,-448 to 10431,-417, blocks 166400,0,-7168 to 166911,255,-6657)
 Actual block type: ID #426 (tile.lotr:armorStand // lotr.common.block.LOTRBlockArmorStand)
 Actual block data value: 1 / 0x1 / 0b0001
Stacktrace:
 at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147549_a(SourceFile:100)
 at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147544_a(SourceFile:92)
 at net.minecraft.client.renderer.RenderGlobal.func_147589_a(RenderGlobal.java:483)
 at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1224)
-- Affected level --
Details:
 Level name: MpServer
 All players: 1 total; [EntityClientPlayerMP['spartanmine11'/120691, l='MpServer', x=166682.06, y=113.62, z=-6739.35]]
 Chunk stats: MultiplayerChunkCache: 121, 121
 Level seed: 0
 Level generator: ID 00 - default, ver 1. Features enabled: false
 Level generator options: 
 Level spawn location: World: (0,69,0), Chunk: (at 0,4,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
 Level time: 278116 game time, 278116 day time
 Level dimension: 0
 Level storage version: 0x00000 - Unknown?
 Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
 Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
 Forced entities: 58 total; [LOTREntityBird['Bird'/873477, l='MpServer', x=166747.03, y=109.00, z=-6745.84], EntityChicken['Chicken'/846110, l='MpServer', x=166669.63, y=108.00, z=-6720.53], LOTREntityBird['Bird'/884510, l='MpServer', x=166609.50, y=93.00, z=-6675.50], EntitySquid['Squid'/886063, l='MpServer', x=166740.50, y=52.00, z=-6787.50], LOTREntityRabbit['Rabbit'/878378, l='MpServer', x=166712.50, y=105.00, z=-6704.50], LOTREntityBird['Bird'/879421, l='MpServer', x=166740.25, y=101.12, z=-6712.75], EntityChicken['Chicken'/866878, l='MpServer', x=166630.63, y=103.00, z=-6676.63], EntitySquid['Squid'/886064, l='MpServer', x=166743.50, y=52.00, z=-6787.50], LOTREntityRabbit['Rabbit'/884274, l='MpServer', x=166753.50, y=106.00, z=-6728.16], LOTREntityRabbit['Rabbit'/882501, l='MpServer', x=166737.50, y=108.00, z=-6742.50], LOTREntityBird['Bird'/883783, l='MpServer', x=166755.18, y=126.77, z=-6692.41], LOTREntityRabbit['Rabbit'/879425, l='MpServer', x=166719.50, y=103.00, z=-6711.50], EntityChicken['Chicken'/866880, l='MpServer', x=166610.56, y=94.00, z=-6682.03], LOTREntityRabbit['Rabbit'/886080, l='MpServer', x=166752.50, y=103.00, z=-6701.50], LOTREntityBird['Bird'/873823, l='MpServer', x=166745.50, y=110.00, z=-6684.50], LOTREntityRabbit['Rabbit'/878427, l='MpServer', x=166725.53, y=105.00, z=-6751.25], LOTREntityRabbit['Rabbit'/878426, l='MpServer', x=166729.41, y=106.00, z=-6749.50], LOTREntityBird['Bird'/882006, l='MpServer', x=166742.57, y=106.05, z=-6725.45], LOTREntityRabbit['Rabbit'/882030, l='MpServer', x=166737.34, y=101.00, z=-6721.63], LOTREntityRabbit['Rabbit'/882027, l='MpServer', x=166730.50, y=107.00, z=-6737.50], LOTREntityBird['Bird'/884348, l='MpServer', x=166752.75, y=109.00, z=-6692.75], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/120693, l='MpServer', x=166662.50, y=112.00, z=-6754.50], EntityClientPlayerMP['spartanmine11'/120691, l='MpServer', x=166682.06, y=113.62, z=-6739.35], EntityItem['item.item.egg'/868216, l='MpServer', x=166654.78, y=110.13, z=-6704.34], EntitySquid['Squid'/873080, l='MpServer', x=166745.94, y=52.22, z=-6789.50], EntityGate['entity.AncientWarfareStructure.aw_gate.name'/120694, l='MpServer', x=166662.50, y=112.00, z=-6756.50], EntitySquid['Squid'/873077, l='MpServer', x=166741.50, y=51.75, z=-6789.78], EntitySquid['Squid'/873079, l='MpServer', x=166739.13, y=51.00, z=-6789.63], LOTREntityBird['Bird'/881292, l='MpServer', x=166746.22, y=112.77, z=-6696.25], LOTREntityBird['Bird'/884381, l='MpServer', x=166732.60, y=101.76, z=-6706.27], LOTREntityBird['Bird'/868249, l='MpServer', x=166758.51, y=127.23, z=-6684.96], LOTREntityBird['Bird'/879012, l='MpServer', x=166737.69, y=108.12, z=-6700.16], LOTREntityBird['Bird'/881596, l='MpServer', x=166712.25, y=106.41, z=-6715.47], LOTREntityBird['Bird'/871359, l='MpServer', x=166729.00, y=101.00, z=-6708.13], LOTREntityButterfly['Butterfly'/885950, l='MpServer', x=166749.91, y=109.53, z=-6691.09], LOTREntityBird['Bird'/882356, l='MpServer', x=166625.50, y=103.00, z=-6695.50], LOTREntityRabbit['Rabbit'/884656, l='MpServer', x=166727.50, y=103.00, z=-6722.50], LOTREntityRabbit['Rabbit'/885938, l='MpServer', x=166749.50, y=106.00, z=-6728.50], EntityChicken['Chicken'/833740, l='MpServer', x=166661.47, y=109.00, z=-6690.53], EntityItem['item.item.egg'/868297, l='MpServer', x=166694.94, y=106.13, z=-6715.44], LOTREntityButterfly['Butterfly'/867797, l='MpServer', x=166732.25, y=115.00, z=-6681.86], EntityCow['Cow'/868049, l='MpServer', x=166623.44, y=98.00, z=-6662.91], EntityCow['Cow'/868048, l='MpServer', x=166609.09, y=92.00, z=-6665.47], LOTREntityBird['Bird'/885714, l='MpServer', x=166723.50, y=103.00, z=-6717.50], LOTREntityButterfly['Butterfly'/868074, l='MpServer', x=166754.30, y=121.95, z=-6678.86], LOTREntityBird['Bird'/881892, l='MpServer', x=166728.87, y=119.95, z=-6712.90], LOTREntityBird['Bird'/867041, l='MpServer', x=166749.03, y=113.73, z=-6696.87], LOTREntityBird['Bird'/869857, l='MpServer', x=166755.80, y=103.56, z=-6717.28], EntityBat['Bat'/869344, l='MpServer', x=166660.55, y=48.51, z=-6738.88], LOTREntityButterfly['Butterfly'/878307, l='MpServer', x=166746.59, y=111.35, z=-6683.99], EntityCow['Cow'/867042, l='MpServer', x=166651.03, y=112.00, z=-6697.84], EntityChicken['Chicken'/833784, l='MpServer', x=166688.47, y=104.00, z=-6704.56], LOTREntityIronHillsMerchant['Kíli son of Khultur, the Iron Hills Merchant'/848880, l='MpServer', x=166733.66, y=104.00, z=-6727.50], LOTREntityDwarf['Khulturg son of Bombur, the Dwarf'/848881, l='MpServer', x=166725.50, y=107.00, z=-6734.75], LOTREntityRabbit['Rabbit'/878576, l='MpServer', x=166757.78, y=107.00, z=-6693.59], LOTREntityDwarf['Fíli son of Gró, the Dwarf'/848882, l='MpServer', x=166726.63, y=107.00, z=-6734.75], LOTREntityDwarf['Farin son of Falin, the Dwarf'/848883, l='MpServer', x=166726.50, y=107.00, z=-6733.53], LOTREntityRabbit['Rabbit'/883186, l='MpServer', x=166613.50, y=96.00, z=-6684.50]]
 Retry entities: 0 total; []
 Server brand: fml,forge
 Server type: Integrated singleplayer server
Stacktrace:
 at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:373)
 at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2444)
 at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:919)
 at net.minecraft.client.main.Main.main(SourceFile:148)
 at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.lang.reflect.Method.invoke(Method.java:483)
 at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
 at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
-- System Details --
Details:
 Minecraft Version: 1.7.10
 Operating System: Windows 7 (amd64) version 6.1
 Java Version: 1.8.0_25, Oracle Corporation
 Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
 Memory: 1515768 bytes (1 MB) / 1060372480 bytes (1011 MB) up to 1060372480 bytes (1011 MB)
 JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx1G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
 AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
 IntCache: cache: 0, tcache: 64173, allocated: 3, tallocated: 727
 FML: MCP v9.05 FML v7.10.99.99 Minecraft Forge 10.13.4.1448 35 mods loaded, 35 mods active
 States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
 UCHIJAAAAAAAAAAAAAA mcp{9.05} [Minecraft Coder Pack] (minecraft.jar) 
 UCHIJAAAAAAAAAAAAAA FML{7.10.99.99} [Forge Mod Loader] (forge-1.7.10-10.13.4.1448-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA Forge{10.13.4.1448} [Minecraft Forge] (forge-1.7.10-10.13.4.1448-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA CodeChickenCore{1.0.4.29} [CodeChicken Core] (minecraft.jar) 
 UCHIJAAAAAAAAAAAAAA NotEnoughItems{1.0.3.74} [Not Enough Items] (Not-Enough-Items-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA lotr{Beta v24.4 for Minecraft 1.7.10} [The Lord of the Rings Mod] ([1.7.10] The Lord of the Rings Mod B24.4.jar) 
 UCHIJAAAAAAAAAAAAAA MCP_API{1.4} [MC+ API] ([1.7.10][1.4] MC+ API.jar) 
 UCHIJAAAAAAAAAAAAAA MCP_Wheel{1.1.2} [MC+ Wheel] ([1.7.10][1.4] MC+ Wheel - v1.1.2.jar) 
 UCHIJAAAAAAAAAAAAAA MCP_Cart{1.3.0} [MC+ Cart] ([1.7.10][1.4] MC+ Cart - v1.3.0.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfare{2.4.83-beta-MC1.7.10} [Ancient Warfare Core] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfareAutomation{2.4.83-beta-MC1.7.10} [Ancient Warfare Automation] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfareNEIPlugin{2.4.83-beta-MC1.7.10} [Ancient Warfare NEI Plugin] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfareNpc{2.4.83-beta-MC1.7.10} [Ancient Warfare NPCs] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfareStructure{2.4.83-beta-MC1.7.10} [Ancient Warfare Structures] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA AncientWarfareVehicle{2.4.83-beta-MC1.7.10} [Ancient Warfare Vehicles] (ancientwarfare-2.4.83-beta-MC1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA animalbikes{1.7.10} [Animal Bikes] (AnimalBikes_1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA BiblioCraft{1.7.3} [BiblioCraft] (BiblioCraft-Mod-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA examplemod{1.0} [examplemod] (colorfularmor-1.7.10-2.1.jar) 
 UCHIJAAAAAAAAAAAAAA colorfularmor{1.7.10-2.1} [Colorful Armor] (colorfularmor-1.7.10-2.1.jar) 
 UCHIJAAAAAAAAAAAAAA DamageIndicatorsMod{3.2.0} [Damage Indicators] (Damage-Indicators-Mod-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DirtToDiamondsMod{V1.0.2} [DirtToDiamondsMod] (Dirt To Diamonds Mod V1.0.2 for Minecraft 1.7.10.zip) 
 UCHIJAAAAAAAAAAAAAA DynamicLights{1.3.7} [Dynamic Lights] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_onFire{1.0.4} [Dynamic Lights Burning Entity Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_creepers{1.0.4} [Dynamic Lights Creeper Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_dropItems{1.0.7} [Dynamic Lights EntityItem Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_entityClasses{1.0.1} [Dynamic Lights Entity Light Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_mobEquipment{1.0.7} [Dynamic Lights Mob Equipment Light Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_flameArrows{1.0.0} [Dynamic Lights Fiery Arrows Light Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_floodLights{1.0.1} [Dynamic Lights Flood Light] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_otherPlayers{1.0.7} [Dynamic Lights OtherPlayers Light Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA DynamicLights_thePlayer{1.1.3} [Dynamic Lights Player Light Module] (DynamicLights-1.7.10.jar) 
 UCHIJAAAAAAAAAAAAAA RadixCore{2.0.2} [RadixCore] (RadixCore-2.0.2-1.7.10-universal.jar) 
 UCHIJAAAAAAAAAAAAAA MCA{5.0.6} [Minecraft Comes Alive] (MCA-5.0.6-1.7.10-universal.zip) 
 UCHIJAAAAAAAAAAAAAA cfm{3.4.8} [§9MrCrayfish's Furniture Mod] (MrCrayfishFurnitureModv3.4.8(1.7.10).jar) 
 UCHIJAAAAAAAAAAAAAA andrew_spartanmod{1.0.1} [SpartanMod] (spartanmod-1.0.1.jar) 
 GL info: ' Vendor: 'ATI Technologies Inc.' Version: '3.3.9836 Compatibility Profile Context' Renderer: 'ATI Radeon HD 4200 '
 Launched Version: 1.7.10-Forge10.13.4.1448-1.7.10
 LWJGL: 2.9.1
 OpenGL: ATI Radeon HD 4200  GL version 3.3.9836 Compatibility Profile Context, ATI Technologies Inc.
 GL Caps: Using GL 1.3 multitexturing.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Anisotropic filtering is supported and maximum anisotropy is 16.
Shaders are available because OpenGL 2.1 is supported.
 Is Modded: Definitely; Client brand changed to 'fml,forge'
 Type: Client (map_client.txt)
 Resource Packs: []
 Current Language: English (US)
 Profiler Position: N/A (disabled)
 Vec3 Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
 Anisotropic Filtering: Off (1)
```

**Comment by alray2569** [comment]
In the meantime, just don't put our armor on an armorstand.

**Comment by alray2569** [comment]
&lt;Note to me>

**Comment by alray2569** [comment]
This may be fixed. Waiting on the beta tester who has the setup right to test this.

**Comment by alray2569** [comment]
Resolved as of Version 1.0.1.2b.
